bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/cache/TTSResult.java

```diff
@@ -0,0 +1,456 @@
+                                              // filechannel
+                try {
+                    FileChannel fileChannelFinal = fileChannel;
+                    if (fileChannelFinal != null) {
+                        try {
+                            logger.debug("Effectively close the TTS cache filechannel for {}", soundFile.getName());
+                            fileChannelFinal.close();
+                        } finally {
+                            fileChannel = null;
+                        }
+                    }
+                } finally {
+                    AudioStream ttsAudioStreamFinal = ttsAudioStream;
+                    if (ttsAudioStreamFinal != null) {
+                        ttsAudioStreamFinal.close();
+                    }
+                }
+            }
+        } finally {
+            fileOperationLock.unlock();
+        }
+    }
+
+    /**
+     * This method is called when we can't defer anymore the call to the TTS service.
+     * The request is then resolved to get the real AudioStream
+     *
+     * @throws TTSException
+     */
+    private void resolveAudioStreamSupplier() throws TTSException {
+        AudioStreamSupplier audioStreamSupplierFinal = ttsAudioStreamSupplier;
+        if (ttsAudioStream == null && audioStreamSupplierFinal != null && !audioStreamSupplierFinal.isResolved()) {
+            logger.trace("Trying to synchronize for resolving supplier");
+            synchronized (audioStreamSupplierFinal) {
+                if (!audioStreamSupplierFinal.isResolved()) { // test again after getting the lock
+                    try {
+                        AudioStream audioStreamResolved = audioStreamSupplierFinal.resolve();
+                        ttsAudioStream = audioStreamResolved;
+                        AudioFormat audioFormatFromTTSAudioStream = audioStreamResolved.getFormat();
+                        this.audioFormat = audioFormatFromTTSAudioStream;
+                        // now that we get the real response format, we can create the .info file
+                        createInfoFile(audioFormatFromTTSAudioStream);
+                    } catch (IOException e) {
+                        throw new TTSException("Cannot create TTS cache file", e);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Get AudioFormat for this TTSResult.
+     * Check the underlying AudioStream if the metadata is not available
+     *
+     * @return
+     */
+    protected AudioFormat getAudioFormat() {
+        AudioFormat audioFormatFinal = this.audioFormat;
+        if (audioFormatFinal != null) {
+            return audioFormatFinal;
+        } else {
+            try {
+                resolveAudioStreamSupplier();
+            } catch (TTSException e) {
+                logger.warn("Cannot get or store audio format from the TTS audio service: {}", e.getMessage());
+            }
+            audioFormatFinal = audioFormat;
```

Comment:
If it's `final`, don't reuse it. Either create another object or re-name to something that is not `final` (e.g. `local`)

Reply:
Indeed. Done by renaming it in "local"